bigtop-deploy/puppet/modules/kerberos/manifests/init.pp

```diff
@@ -189,8 +193,9 @@
     $keytab = "/etc/$title.keytab"
 
     $internal_princs = $spnego ? {
-      /(true|enabled)/ => [ 'HTTP' ],
-      default          => [ ],
+      true      => [ 'HTTP' ],
+      'enabled' => [ 'HTTP' ],
+      default   => [ ],
```

Comment:
The original code won't work hence you refactor here?

Reply:
Yes. The credential of HTTP/host@REALM did not written to keytab file since boolean `true` was not covered in the conditional of previous code.